src/docgen/render/markdown.ts

```diff
@@ -233,3 +220,46 @@ export class Markdown {
     return line.replace(/\|/g, '\\|');
   }
 }
+
+export const defaultAnchorFormatter = (type: JsiiEntity) => {
+  return type.id;
+};
+
+export const defaultLinkFormatter = (type: JsiiEntity) => {
+  const name = type.fqn.split('.').pop()!;
+  return `<a href="#${type.id}">${name}</a>`;
```

Comment:
What if the type comes from a different assembly?

Reply:
In that case we just shrug it off. This is just a default after all. The id is unique across assemblies, so in the worst case the link is invalid. You can see this in the example gist [here](https://gist.github.com/Chriscbr/d0833a9ecce258dead648a63253b8b0e), where if you click on a type named `Construct`, the link doesn't take you anywhere.

I think this is good enough for the default.

Reply:
Actually, I suppose we could display no link if the type isn't from the current assembly. I'll leave this as a future enhancement for now since it's not too critical

Reply:
See I disagree with this.

A link that doesn't work seems worse than no link at all. I'd rather have that be in the initial implementation.

Reply:
Got it, I see your point now - yeah dead links are a pretty bad experience. 😅

I've added another parameter to the `linkFormatter` API to support this -- as of right now, `linkFormatter` is expected to be a pure function, so the formatter needs extra info to make the determination of whether to create a link.

The formatting APIs are getting a bit heavy/awkward so I've added some more docs and marked them experimental for now. It might make sense eventually just require anyone that wants to customize the markdown linking and formatting etc. to just implement their own markdown renderer at the end of the day -- but for now I'd rather keep these (awkward) APIs just to avoid duplicating code between here and Construct Hub.